### PR TITLE
Add deprecation warning for Python 2.6

### LINKED ISSUE #3779: Add deprecation warning for Python 2.6 when importing Astropy
As per the discussion in https://groups.google.com/forum/#!topic/astropy-dev/i71P1Yatml8, for 1.1 we should add a deprecation warning when astropy is imported on Python 2.6.

### PR DESCRIPTION
**Pull request by lingxz**
I hope this is the correct place to add the deprecation warning!

Closes #3779

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1758,6 +1758,8 @@ Other Changes and Additions
   source code from the git repository, in order to allow the ERFA wrappers to
   be generated. [#3166]
 
+- Added deprecation warning for Python 2.6 [#3779].
```
Comment (embray) [inline CHANGES.rst]: I this will probably deserve a little more explanation, but I can add that.

File: astropy/__init__.py [hunk 2]
```diff
@@ -219,6 +219,9 @@ def _initialize_astropy():
     is_astropy_source_dir = os.path.exists(os.path.join(source_dir, os.pardir,
                                                         '.astropy-root'))
 
+    if sys.version_info[:2] < (2,7):
+        warn("Python 2.6 will no longer be supported from Astropy v1.2 onwards.", DeprecationWarning)
```
Comment (embray) [inline astropy/__init__.py]: This should probably still be an `AstropyDeprecationWarning`, unless that causes too many problems at import-time.

**Comment by embray** [comment]
Thanks for the reminder about this.

**Comment by lingxz** [comment]
Changed it to ```AstropyDeprecationWarning```, thanks for the help.

**Comment by embray** [comment]
In the interest of time I'm going to manually merge this and make a few more changes.  Thanks for reminding about this though.

**Comment by lingxz** [comment]
No problem, still pretty new to astropy, so thanks for the patience guiding me through my pull requests.

**Comment by embray** [comment]
@lingxz No thank you for helping out!

**Pull request closed without merging**